src/averageAges.js

```diff
@@ -14,12 +14,39 @@
  *
  * @return {number}
  */
+
+function getAverageFromArr(arr) {
+  return +(arr.reduce((a, b) => a + b, 0) / arr.length).toFixed(2);
+}
+
+const getYearsArr = (arr) => {
+  return arr.map(person => {
+    const { died, born } = person;
+
+    return +died - +born;
+  });
+};
+
 function calculateMenAverageAge(people, century) {
   // write code here
   // learn how to use array methods like .filter .map .some .every .find .reduce
   // avoid using loop and forEach
   // replace `if ()` statement with &&, || or ?:
   // without nesting
+  const manArr = people.filter(person => person.sex === 'm');
+
+  let yearsArr;
+
+  if (century) {
+    const filteredArr = manArr
+      .filter(person => Math.ceil(person.died / 100) === century);
+
+    yearsArr = getYearsArr(filteredArr);
+  } else {
+    yearsArr = getYearsArr(manArr);
+  };
```

Comment:
```suggestion
  const manArr = people.filter(person =>
    century
      ? person.sex === 'm' && Math.ceil(person.died / 100) === century
      : person.sex === 'm'
  );
  
 let yearsArr = getYearsArr(manArr);
```

Use ternary instead of if/else statements.

Fix all such cases